config-store-yaml/src/main/java/io/stargate/config/store/api/yaml/ConfigStoreActivator.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.config.store.api.yaml;
+
+import io.stargate.config.store.api.ConfigStore;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Hashtable;
+import org.osgi.framework.BundleActivator;
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConfigStoreActivator implements BundleActivator {
+
+  private static final Logger logger = LoggerFactory.getLogger(ConfigStoreActivator.class);
+
+  public static final String CONFIG_STORE_YAML_IDENTIFIER = "ConfigStoreYaml";
+
+  private boolean configStoreEnabled;
+
+  private final String configYamlLocation;
+
+  // for testing purpose
+  public ConfigStoreActivator(boolean configStoreEnabled, String configYamlLocation) {
+    this.configStoreEnabled = configStoreEnabled;
+    this.configYamlLocation = configYamlLocation;
+  }
+
+  public ConfigStoreActivator() {
+    this(
+        Boolean.parseBoolean(System.getProperty("stargate.config_store.enabled", "false")),
+        System.getProperty(
+            "stargate.config_store.yaml.location", "/etc/stargate/stargate-config.yaml"));
+  }
+
+  @Override
+  public void start(BundleContext context) {
+    if (!configStoreEnabled) {
+      logger.info("Config Store YAML disabled - it will not be started");
+      return;
+    }
+    logger.info("Starting Config Store YAML...");
+    Path yamlFilePath = Paths.get(configYamlLocation);
+    if (!Files.exists(yamlFilePath)) {
+      throw new IllegalArgumentException(
+          String.format(
+              "The yaml file does not exists, please check the path: %s. The ConfigStoreYaml will not be registered.",
+              yamlFilePath));
+    }
```

Comment:
Sorry to be a pain on this issue, but I think in this situation if/when we have another service depending on the config service, that other service will keep waiting on startup if the config service is not enabled. I believe this situation would be hard to debug due to lack of explicit error messages.

What I'm proposing is to _always_ start and _register_ the config service implementation, but throw exceptions about missing config data when that data is requested in run time. This way configuration errors/mistakes will be reported in the context of the service that actually uses that configuration and the error will be explicit. WDYT?

Reply:
The main reason why I decided to validate that is that the file under `/etc/stargate/stargate-config.yaml"` must always be available when deploying stargate using k8s. The lack of this file is an abnormal situation. 
So if the config-store is enabled (i.e., on production), the file must be present. If it is not, it means that the deployment process is broken - it didn't create a config map.

On the other hand, throwing will make the development harder (starting stargate locally). So postponing this to the runtime will solve this problem, but it also means that we will detect the fact that deployment is broken very late on production. I am not sure if we want such behavior.
Another solution would be to change this:
```
  System.getProperty(
            "stargate.config_store.yaml.location", "/etc/stargate/stargate-config.yaml"));
```
to a location in the stargate repository (it can contain the empty file for now) and change this to `etc/stargate/stargate-config.yaml` when deploying using k8s.

Reply:
Do we expect any other services to use the config in local builds or public docker images?

Reply:
Just to clarify: I'm proposing to throw "missing config" exceptions from `ConfigStoreYaml.getConfigForModule`

Reply:
> What I'm proposing is to always start and register the config service implementation, but throw exceptions about missing config data when that data is requested in run time

I agree with @dimas-b here. If we think about the simplest case of developer-mode here it's perfectly valid to start up stargate without a config file and just a few command line args. In that case everything will register and run just fine. But, lets say you're running in prod with the kafka producer and the file is missing. In that case the _kafka producer_ will cause startup to fail because it's missing config it needs. Yes, it's not 100% explicit as you'll then have to dig through and figure out why the config is missing but it'll be more obvious than not starting configstore or blindly creating a new file.

Reply:
Ok, if we have a quorum here, I am also ok with dropping that extra validation. 
Implemented in: https://github.com/stargate/stargate/pull/305/commits/43ed0fd33eaeca453beb6d4f7a2bf555dc52bad5

@dimas-b The throw missing config works as you described, so there is no need for change there